duecredit/io.py

```diff
@@ -154,6 +154,29 @@ def _model_citations(self, tags=None):
                 target_dict = packages
             target_dict['citations'][path].append(citation)
             target_dict['entry_keys'][path].append(entry_key)
+
+        # now we need to filter out the packages that don't have modules
+        # or objects cited
+        cited_packages = list(packages['citations'].keys())
+        cited_modules = list(modules['citations'].keys())
+        cited_objects = list(objects['citations'].keys())
+
+        for cited_package in cited_packages:
+            children = list(filter(lambda x: x.startswith(cited_package),
+                              cited_modules + cited_objects))
```

Comment:
you could just create a single list `cited = list(modules['citations']) + list(modules['citations'])` and avoid having 2 lists concatenated here all the time.  moreover I think that 'startswith' might be 'insufficient' if `cited_package` name is just a prefix (e.g. `nipy` whenever cited_module would be `nipype`) ;)    in any case you might not need any of those since `Citation` implements `__contains__` so theoretically (didn't check if we have instances here or just names/paths available)  you should be able to do `if citation1 in citation2` which would be True if path of citation1 "under" citation2 path.  If we deal with strings, just extract that logic from `__contains__` into a helper function (e.g. `is_contained(toppath, subpath)`) to do that check and use it in both `__contains__` and here